platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/pipeline/VirtualPipeline.java

```diff
@@ -460,57 +509,34 @@ private void merge(final PipelineListNode<VirtualRoot> node) {
         copy.merge();
     }
 
-    /**
-     * Check if a copy should be removed from the pipeline. Only remove copies when they are at
-     * the end of their lifecycle.
-     */
-    private static boolean shouldBeRemovedFromPipeline(final VirtualRoot copy) {
-        return copy.isDestroyed() && (copy.isFlushed() || copy.isMerged());
-    }
-
-    /**
-     * Check if a copy should prevent newer copies from being flushed.
-     */
-    private static boolean shouldBlockFlushes(final VirtualRoot copy) {
-        return !(copy.isDestroyed() || copy.isDetached())
-                || (copy.shouldBeMerged() && !copy.isMerged())
-                || (copy.shouldBeFlushed() && !copy.isFlushed());
-    }
-
     /**
      * Hash, flush, and merge all copies currently capable of these operations.
      */
     private void hashFlushMerge() {
         PipelineListNode<VirtualRoot> next = copies.getFirst();
-
-        // We can only flush a copy if there exists no older copy that is not
-        // either destroyed or detached. Once we encounter the first that is neither,
-        // all newer copies will be prevented from flushing.
-        boolean flushBlocked = false;
-
-        // iterate from the oldest copy to the newest
-        while (next != null && !Thread.currentThread().isInterrupted()) {
+        // Iterate from the oldest copy to the newest
+        while ((next != null) && !Thread.currentThread().isInterrupted()) {
```

Comment:
checking if the current thread is interrupted clears the interrupted flag.  It's generally recommended to store the result of the call and ensure the thread is re-interrupted or cleanly ends if it was interrupted when the status was checked.

Reply:
`isInterrupted()` doesn't clear the interrupted status (but `interrupted()` does), so it should be fine here.